### ISSUE TITLE
`partner_firstname` is Incompatible with `auth_oauth_signup`

### ISSUE THREAD

**Issue by bwrsandman** [OCA/partner-contact#30]
### Description

When a new user signs in with a configured oauth client on a server which has partner_firstname installed, Odoo will copy the TemplateUser with default values (name, login, email, etc).

Due to the way partner_firstname deals with name to firstname and lastname, it will simply copy the TemplateUser's name are result with a new user with the name "Template User (copy)".

The line which does this is here: https://github.com/OCA/partner-contact/blob/7.0/partner_firstname/res_user.py#L33

This problem could and should be resolved by making an autoinstall module called `auth_oauth_signup_partner_firstname` which alters the call to `signup()` to also have `firstname` and `lastname`.

This could be really easy to implement **with a small change to Odoo's code**. The `auth_oauth_signup` module only adds a single function. This function could be factorized so that the default values creation could be inherited to add the two keys we need without the need to completely overwrite the only function in the entire module.

All that is needed is to factor out the generation of `values` (https://github.com/odoo/odoo/blob/7.0/addons/auth_oauth_signup/res_users.py#L47) from the `validation` field into a function such as `_generate_signup_values()`.

Without this change directly to the `auth_oauth_signup` module, every module changing `res_users` or with a custom oauth implementation will need to essentially rewrite `auth_oauth_signup`. What I propose is to factor out the code which desirable to change.
## With factorization

``` python
from openerp.osv import orm


class res_users(orm.Model):
    _inherit = 'res.users'

    def _generate_signup_values(
            self, cr, uid, provider, validation, params, context=None):
        values = super(res_users, self)._generate_signup_values(
            cr, uid, provider, validation, params, context=context
        )
        values.update(
            lastname=validation.get('lastname') or values.get('name') or False,
            firstname=validation.get('firstname') or False,
        )
        return values
```

The PR to Odoo related to this: https://github.com/odoo/odoo/pull/2355

**Comment by pedrobaeza** [comment]
While we wait for the factorisation, we can simply override method, call super, and then write again in the created id missing values.

**Comment by bwrsandman** [comment]
Fixed upstream, now the 7.0 module needs the update I mentionned

**Comment by NL66278** [comment]
Fixed upstream and 7.0 is long out of support, so closing.
